slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -94,6 +94,12 @@ trait MySQLProfile extends JdbcProfile { profile =>
         else l
       }
     }
+
+    //Reference: https://github.com/slick/slick/issues/1419
+    override def createTableNamer(meta: MTable): TableNamer = new TableNamer(meta){
+      override def schema = meta.name.catalog
+      override def catalog = meta.name.schema 
```

Comment:
@lastland  So only the schema should be set to catalog and catalog left unchanged ?

As far as solving this in Slick and maintaining the uniform, the only way would be change this in MySQLProfile. 

Slick still inteprets this as a schema so this would still produce valid sql query.